packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2704,14 +2710,81 @@ class _Location {
   }
 }
 
+/// Describer to gather important information about [debugCreator].
+///
+/// This function will be attached to [FlutterErrorDetails.describeCreator]
+/// in [WidgetsBinding].
+Iterable<DiagnosticsNode> describeCreator(dynamic debugCreator) {
+  final Element element = debugCreator is Element? debugCreator: debugCreator.element;
+  return <DiagnosticsNode>[
+    _describeRelevantUserCode(element),
+  ].where((DiagnosticsNode node) => node != null);
+}
+
+DiagnosticsNode _describeRelevantUserCode(Element element) {
+  if (!WidgetInspectorService.instance.isWidgetCreationTracked()) {
+    return ErrorDescription(
+      'Widget creation tracking is currently disabled. Enabling'
+      ' it enables improved error messages. It can be enabled by passing '
+      '`--track-widget-creation` to `flutter run` or `flutter test`.',
+    );
+  }
+  ErrorDescription desc;
+  element.visitAncestorElements((Element ancestor) {
+    // TODO(chunhtai): should print out all the widgets that are about to cross
+    // package boundaries.
+    if (_isLocalCreationLocation(ancestor)) {
+      desc = ErrorDescription(
+        'User created ancestor of the error causing '
```

Comment:
This indent seems a bit non standard relative to the other diagnostics. See the other comment on the error expectation describing why. Generally the '\n\n' is a bit odd and indenting the line following a ':' by two spaces is standard.
If you want to use the standard indent, here are two options code that will opt you in and better indicate that the widget and location are linked.
```dart
DiagnosticsBlock(
  name: 'User created ancestor of the error causing widget was',
  children: <DiagnosticsNode>[
    ancestor.widget.toDiagnosticsNode(),
   _describeCreationLocation(ancestor),
  ],
);
```
This will give you output:
```
User created ancestor of the error causing widget was:
  MyWidget(relevantProperty1: relevantValue1, ..)
  some/path/to/file.dart
```

or 
If you want to use the standard indent, here are two options code that will opt you in and better indicate that the widget and location are linked.
```dart
DiagnosticsBlock(
  name: 'User created ancestor of the error causing widget was',
  children: <DiagnosticsNode>[
    ancestor.widget.toDiagnosticsNode(),
   _describeCreationLocation(ancestor),
  ],
);
```
This will give you output:
```
User created ancestor of the error causing widget was:
  MyWidget(relevantProperty1: relevantValue1, ..)
  some/path/to/file.dart
```
I thought this was what you wanted from the expectation file but I think that file just had an extra line break that you didn't want.

If you want to keep the widget on its own line as you have currently, I think this would also be fine:
```dart
DiagnosticsBlock(
  name: 'User created ancestor of the error causing widget was',
  children: <DiagnosticsNode>[
    ErrorDescription('$ancestor.widget.toStringShort()} ${_describeCreationLocation(ancestor)'}),
  ],
);
```
This will give you output:
```
User created ancestor of the error causing widget was:
  MyWidget some/path/to/file.dart
```

Reply:
I will go with the one line because in the future we will want to print out multiple widget, pretty much like the stack trace. One line will looks more clean.